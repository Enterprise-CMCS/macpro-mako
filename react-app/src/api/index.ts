export * from "./useSearch";
export * from "./useGetForm";
export * from "./useGetItem";
export * from "./useGetUser";
export * from "./getAttachmentUrl";
export * from "./useGetPackageActions";
export * from "./useGetCPOCs";
export * from "./useGetSystemNotifs";
export * from "./useGetTypes";
export * from "./amplifyConfig";
export * from "./itemExists";
export * from "./useGetCounties";
