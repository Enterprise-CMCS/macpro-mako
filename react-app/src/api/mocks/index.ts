export * as mockSubmit from "./submit";
export * as mockTypes from "./types";
