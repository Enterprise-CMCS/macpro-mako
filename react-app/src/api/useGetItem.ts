import { useQuery, UseQueryOptions } from "@tanstack/react-query";
import { API } from "aws-amplify";
import { opensearch, ReactQueryApiError, SEATOOL_STATUS } from "shared-types";

import { sendGAEvent } from "@/utils/ReactGA/SendGAEvent";

export const getItem = async (id: string): Promise<opensearch.main.ItemResult> =>
  await API.post("os", "/item", { body: { id } }).catch(() =>
    sendGAEvent("api_error", { message: `failure /item ${id}` }),
  );

export const idIsApproved = async (id: string) => {
  try {
    const record = await getItem(id);
    return record._source.seatoolStatus == SEATOOL_STATUS.APPROVED;
  } catch (e) {
    console.error(e);
    return false;
  }
};

export const canBeRenewedOrAmended = async (id: string) => {
  try {
    const record = await getItem(id);
    return ["New", "Renew"].includes(record._source.actionType);
  } catch (e) {
    console.error(e);
    return false;
  }
};

export const useGetItem = (
  id: string,
  options?: UseQueryOptions<opensearch.main.ItemResult, ReactQueryApiError>,
) => {
  return useQuery<opensearch.main.ItemResult, ReactQueryApiError>(
    ["record", id],
    () => getItem(id),
    options,
  );
};
