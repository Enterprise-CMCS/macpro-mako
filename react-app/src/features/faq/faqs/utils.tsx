import { cn } from "@/utils";

export interface Template {
  title: string;
  text?: string;
  href: string;
  subtext?: string[];
}

export const helpDeskContact = {
  email: "OneMAC_Helpdesk@cms.hhs.gov",
  phone: "(833) 228-2540",
};

export const slugify = (text: string) =>
  text
    .toLowerCase()
    .replace(/[^a-z0-9]+/g, "-")
    .replace(/(^-|-$)/g, "")
    .substring(0, 40);

export const handleSupportLinkClick = (type: string) => (e: React.MouseEvent<HTMLElement>) => {
  const text = e.currentTarget.textContent?.trim() || "unknown";
  window.gtag("event", `support_click_${type}_${slugify(text)}`, {
    event_category: "Support",
    event_label: text,
  });
};

export const PdfLink = ({
  href,
  title,
  text,
  label,
  className = "underline hover:no-underline",
}: {
  href: string;
  title: string;
  text?: string;
  label: string;
  className?: string;
}) => (
  <a
    className={cn("text-primary", className)}
    href={href}
    download
    rel="noopener noreferrer"
    target="_blank"
    onClick={handleSupportLinkClick(label)}
  >
    {title}
    {text && `: ${text}`}
  </a>
);

export const PdfList = ({
  list,
  label,
  ulClassName = "list-disc pl-6 space-y-2",
}: {
  list: Template[];
  label: string;
  ulClassName?: string;
}) => (
  <ul className={ulClassName} role="list">
    {list.map((pdf) => (
      <li key={pdf.title}>
        <PdfLink href={pdf.href} title={pdf.title} text={pdf.text} label={label} className="" />
        {pdf.subtext && (
          <ul className="list-disc pl-7 space-y-1" role="list">
            {pdf.subtext.map((sub, index) => (
              <li key={index} className="text-sm text-gray-600">
                {sub}
              </li>
            ))}
          </ul>
        )}
      </li>
    ))}
  </ul>
);
