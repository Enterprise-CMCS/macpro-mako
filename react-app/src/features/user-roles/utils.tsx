export type StatusType = "active" | "pending" | "denied" | "revoked";
export type UserRoleType = {
  id: string;
  email: string;
  fullName: string;
  role: string;
  territory: string;
  doneByEmail: string;
  doneByName: string;
  lastModifiedDate: number;
  status: StatusType;
  eventType: string;
  group?: string;
  division?: string;
};

export const initSortUserData = (userData: UserRoleType[]) => {
  if (!userData.length) return [];
  // separate pending / other
  const pendingRoles = userData.filter((x: UserRoleType) => x.status === "pending");
  const remainingRoles = userData.filter((x: UserRoleType) => x.status !== "pending");

  const compare = (a: UserRoleType, b: UserRoleType) => {
    const nameA = a.fullName.toLocaleLowerCase();
    const nameB = b.fullName.toLocaleLowerCase();

    if (nameA < nameB) return -1;
    if (nameA > nameB) return 1;
    return 0;
  };
  const sorted = pendingRoles.sort(compare).concat(remainingRoles.sort(compare));

  return sorted;
};

export const sortUserData = (
  sortByKey: keyof UserRoleType,
  direction: boolean,
  data: UserRoleType[],
) => {
  if (!data.length) return [];

  // when direction is true, that means we are descending
  const [last, first] = direction ? [-1, 1] : [1, -1];
  const sortStatus = (a: UserRoleType, b: UserRoleType) => {
    switch (a.status) {
      case "pending":
        return first;
      case "active":
        return b.status === "pending" ? last : first;
      case "denied":
        return b.status === "revoked" ? first : last;
      case "revoked":
        return last;
      default:
        return last;
    }
  };
  const mainSort = (a: UserRoleType, b: UserRoleType) =>
    a[sortByKey] < b[sortByKey] ? first : last;

  if (sortByKey === "status") return data.sort((a, b) => sortStatus(a, b));

  return data.sort((a, b) => mainSort(a, b));
};
