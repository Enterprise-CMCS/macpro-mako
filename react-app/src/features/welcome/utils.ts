import { QueryClient } from "@tanstack/react-query";
import { redirect } from "react-router";

import { getUser } from "@/api";

export const loader = (queryClient: QueryClient) => {
  return async () => {
    const queryString = window.location.search;
    // Parse the query string to get URL parameters
    const queryParams = new URLSearchParams(queryString);
    // Access specific parameters
    const errorDescription = queryParams.get("error_description");
    const error = queryParams.get("error");
    if (errorDescription || error) {
      console.error("Authentication Error:", { errorDescription, error });
      return { error };
    }

    // check user query has been initialized
    if (!queryClient.getQueryData(["user"])) {
      const userFetch = await queryClient.fetchQuery({
        queryKey: ["user"],
        queryFn: () => getUser(),
      });
      const isUserLoggedIn =
        !["/login", "/faq"].includes(window.location.pathname) && !userFetch?.user;

      if (isUserLoggedIn) {
        return redirect("/login");
      }
    }

    return queryClient.getQueryData(["user"]);
  };
};
