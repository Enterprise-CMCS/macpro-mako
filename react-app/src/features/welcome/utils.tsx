import { QueryClient } from "@tanstack/react-query";
import { Navigate, redirect } from "react-router";

import { getUser, OneMacUser } from "@/api";

export const loader = (queryClient: QueryClient) => {
  return async () => {
    const queryString = window.location.search;
    // Parse the query string to get URL parameters
    const queryParams = new URLSearchParams(queryString);
    // Access specific parameters
    const errorDescription = queryParams.get("error_description");
    const error = queryParams.get("error");
    if (errorDescription || error) {
      console.error("Authentication Error:", { errorDescription, error });
      return { error };
    }

    // check user query has been initialized
    if (!queryClient.getQueryData(["user"])) {
      const userFetch = await queryClient.fetchQuery({
        queryKey: ["user"],
        queryFn: () => getUser(),
      });
      return !userFetch?.user && redirect("/login");
    }

    // check user is logged in
    const loginRedirect =
      !["/login", "/faq", "/support"].includes(window.location.pathname) &&
      !queryClient.getQueryData<OneMacUser>(["user"])?.user;
    if (loginRedirect) {
      return <Navigate to="/login" />;
    }

    return queryClient.getQueryData(["user"]);
  };
};
