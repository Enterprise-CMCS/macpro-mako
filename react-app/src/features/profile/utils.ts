import { UserRole } from "shared-types/events/legacy-user";

import { StateAccess } from "@/api";
import { convertStateAbbrToFullName } from "@/utils";

export const userRoleMap = {
  defaultcmsuser: "CMS Read-only User",
  cmsroleapprover: "CMS Role Approver",
  cmsreviewer: "CMS Read-only User",
  statesystemadmin: "State System Admin",
  helpdesk: "Helpdesk",
  statesubmitter: "State Submitter",
  systemadmin: "System Admin",
};

export const stateAccessRoles: UserRole[] = [
  "statesubmitter",
  "statesystemadmin",
  "cmsroleapprover",
  "systemadmin",
];

export const orderStateAccess = (accesses: StateAccess[]) => {
  if (!accesses || !accesses.length) return;
  // sort revoked states seprately and add to
  const activeStates = accesses.filter((x: StateAccess) => x.status != "revoked");
  const revokedStates = accesses.filter((x: StateAccess) => x.status == "revoked");

  const compare = (a: StateAccess, b: StateAccess) => {
    const stateA = convertStateAbbrToFullName(a.territory);
    const stateB = convertStateAbbrToFullName(b.territory);

    if (stateA < stateB) return -1;
    if (stateA > stateB) return 1;
    return 0;
  };

  const sorted = activeStates.sort(compare).concat(revokedStates.sort(compare));

  return sorted;
};

// if user has no active roles, show pending state(s)
// show state(s) for latest active role
export const filterStateAccess = (userDetails, userProfile) => {
  if (!userProfile?.stateAccess || userProfile.stateAccess.length < 1) return [];
  return userDetails?.role
    ? userProfile.stateAccess.filter(
        (access: StateAccess) => access.role === userDetails.role && access.territory !== "ZZ",
      )
    : userProfile.stateAccess.filter((access: StateAccess) => access.territory !== "ZZ");
};
