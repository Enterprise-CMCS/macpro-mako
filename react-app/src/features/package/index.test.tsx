import { describe, expect, it, vi } from "vitest";
import { UseQueryResult } from "@tanstack/react-query";
import * as api from "@/api/useGetUser";

import { DetailsContent } from ".";
import { mockUseGetUser, ADMIN_CHANGE_ITEM, ADMIN_ITEM_ID } from "mocks";
import { OneMacUser } from "@/api/useGetUser";
import { renderFormAsync } from "@/utils/test-helpers";
import * as gi from "@/api/useGetItem";
describe("package details", () => {
  vi.spyOn(gi, "useGetItemCache").mockReturnValue({
    data: ADMIN_CHANGE_ITEM._source,
    refetch: vi.fn(),
  });

  it("makes a package", async () => {
    vi.spyOn(api, "useGetUser").mockImplementation(() => {
      const response = mockUseGetUser();
      return response as UseQueryResult<OneMacUser, unknown>;
    });
    const { asFragment } = await renderFormAsync(<DetailsContent id={ADMIN_ITEM_ID} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
