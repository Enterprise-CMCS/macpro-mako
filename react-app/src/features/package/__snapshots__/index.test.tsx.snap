// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`package details > makes a package 1`] = `
<DocumentFragment>
  <div
    class="w-full py-1 px-4 lg:px-8"
  >
    <section
      class="block md:flex space-x-0 md:space-x-8"
      id="package_overview"
    >
      <div
        class="my-6 sticky top-12"
      >
        <div
          class="h-2 shadow-lg"
          style="border-radius: 3px 3px 0px 0px; height: 8px;"
        />
        <div
          class="border border-t-0 rounded-b-sm border-slate-300"
        >
          <div
            class="p-4 py-1 w-80 min-h-36"
          >
            <h2>
              Status
            </h2>
            <div
              class="my-3 max-w-2xl font-bold text-xl"
            >
              <div />
              <div
                class="flex mt-1 flex-col gap-1 items-start"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="my-6 sticky top-12"
      >
        <div
          class="h-2 shadow-lg"
          style="border-radius: 3px 3px 0px 0px; height: 8px;"
        />
        <div
          class="border border-t-0 rounded-b-sm border-slate-300"
        >
          <div
            class="p-4 py-1 w-80 min-h-36"
          >
            <h2>
              Package Actions
            </h2>
            <div
              class="my-3"
            >
              <em
                class="text-gray-400 my-3"
              >
                No actions are currently available for this submission.
              </em>
            </div>
          </div>
        </div>
      </div>
    </section>
    <div
      class="flex flex-col gap-3"
    >
      <section
        class="p-4 border rounded-sm border-cardBorder"
        id="package_details"
      >
        <h1
          class="text-3xl font-semibold mb-2"
          data-testid="undefined-title"
        >
          undefined Package Details
        </h1>
        <hr
          class="my-6 bg-slate-200"
        />
        <div
          class="gap-8 flex flex-col"
          data-testid="undefined-child"
        >
          <div
            class="flex-col gap-4 max-w-2xl"
          >
            <div
              class="grid grid-cols-2 gap-6 py-4"
            >
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Submission ID
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  SS-1235.R00.01
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Authority
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                />
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  State
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  Maryland
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Proposed effective date
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  Pending
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Initial submission date
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  -- --
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Latest package activity
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  -- --
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Formal RAI response date
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  -- --
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Final disposition date
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  -- --
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Approved effective date
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  -- --
                </div>
              </div>
            </div>
            <hr
              class="my-4"
            />
            <div
              class="grid grid-cols-2 gap-6"
            >
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  Submitted by
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  <p
                    class="text-lg"
                  >
                    -- --
                  </p>
                </div>
              </div>
              <div>
                <h3
                  style="font-weight: 700;"
                >
                  CPOC
                </h3>
                <div
                  class="py-2"
                  style="font-weight: 400;"
                >
                  <p
                    class="text-lg"
                  >
                    -- --
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="p-4 border rounded-sm border-cardBorder"
        id="administrative_package_changes"
      >
        <h1
          class="text-3xl font-semibold mb-2"
          data-testid="undefined-title"
        >
          Administrative Package Changes (6)
        </h1>
        <hr
          class="my-6 bg-slate-200"
        />
        <div
          class="gap-8 flex flex-col"
          data-testid="undefined-child"
        >
          <p
            class="text-sm"
          >
            Administrative changes reflect updates to specific data fields. If you have additional questions, please contact the assigned CPOC.
          </p>
          <div
            class="flex flex-col gap-2"
            data-orientation="vertical"
          >
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="open"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="open"
              >
                <button
                  aria-controls="radix-:r1:"
                  aria-expanded="true"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180 bg-gray-100 px-3"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="open"
                  id="radix-:r0:"
                  type="button"
                >
                  <p
                    class="flex flex-row gap-2 text-gray-600"
                  >
                    <strong>
                      Package Added
                    </strong>
                     - Sun, Jan 1, 2023 12:00:00 AM
                  </p>
                  <svg
                    class="h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:r0:"
                class="text-sm transition-all p-4"
                data-orientation="vertical"
                data-state="open"
                id="radix-:r1:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width); transition-duration: 0s; animation-name: none;"
              >
                <div
                  class="pb-4 pt-0"
                >
                  <div
                    class="flex flex-col gap-6"
                  >
                    <div>
                      <h2
                        class="font-bold text-lg mb-2"
                      >
                        Change Made
                      </h2>
                      <p>
                        add file
                      </p>
                    </div>
                    <div>
                      <h2
                        class="font-bold text-lg mb-2"
                      >
                        Change Reason
                      </h2>
                      <p>
                        missing file
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="closed"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="closed"
              >
                <button
                  aria-controls="radix-:r3:"
                  aria-expanded="false"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180 bg-gray-100 px-3"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="closed"
                  id="radix-:r2:"
                  type="button"
                >
                  <p
                    class="flex flex-row gap-2 text-gray-600"
                  >
                    <strong>
                      Package Added
                    </strong>
                     - Sun, Jan 1, 2023 12:00:00 AM
                  </p>
                  <svg
                    class="h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:r2:"
                class="text-sm transition-all p-4"
                data-orientation="vertical"
                data-state="closed"
                hidden=""
                id="radix-:r3:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              />
            </div>
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="closed"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="closed"
              >
                <button
                  aria-controls="radix-:r5:"
                  aria-expanded="false"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180 bg-gray-100 px-3"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="closed"
                  id="radix-:r4:"
                  type="button"
                >
                  <p
                    class="flex flex-row gap-2 text-gray-600"
                  >
                    <strong>
                      Manual Update
                    </strong>
                     - Sun, Jan 1, 2023 12:00:00 AM
                  </p>
                  <svg
                    class="h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:r4:"
                class="text-sm transition-all p-4"
                data-orientation="vertical"
                data-state="closed"
                hidden=""
                id="radix-:r5:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              />
            </div>
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="closed"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="closed"
              >
                <button
                  aria-controls="radix-:r7:"
                  aria-expanded="false"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180 bg-gray-100 px-3"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="closed"
                  id="radix-:r6:"
                  type="button"
                >
                  <p
                    class="flex flex-row gap-2 text-gray-600"
                  >
                    <strong>
                      -- --
                    </strong>
                     - Sun, Jan 1, 2023 12:00:00 AM
                  </p>
                  <svg
                    class="h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:r6:"
                class="text-sm transition-all p-4"
                data-orientation="vertical"
                data-state="closed"
                hidden=""
                id="radix-:r7:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              />
            </div>
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="closed"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="closed"
              >
                <button
                  aria-controls="radix-:r9:"
                  aria-expanded="false"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180 bg-gray-100 px-3"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="closed"
                  id="radix-:r8:"
                  type="button"
                >
                  <p
                    class="flex flex-row gap-2 text-gray-600"
                  >
                    <strong>
                      Enable Formal RAI Response Withdraw
                    </strong>
                     - Sun, Jan 1, 2023 12:00:00 AM
                  </p>
                  <svg
                    class="h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:r8:"
                class="text-sm transition-all p-4"
                data-orientation="vertical"
                data-state="closed"
                hidden=""
                id="radix-:r9:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              />
            </div>
            <div
              class="border-b"
              data-orientation="vertical"
              data-state="closed"
            >
              <h3
                class="flex"
                data-orientation="vertical"
                data-state="closed"
              >
                <button
                  aria-controls="radix-:rb:"
                  aria-expanded="false"
                  class="flex flex-1 items-center justify-between py-4 font-medium transition-all hover:underline [&[data-state=open]>svg]:rotate-180 bg-gray-100 px-3"
                  data-orientation="vertical"
                  data-radix-collection-item=""
                  data-state="closed"
                  id="radix-:ra:"
                  type="button"
                >
                  <p
                    class="flex flex-row gap-2 text-gray-600"
                  >
                    <strong>
                      Disable Formal RAI Response Withdraw
                    </strong>
                     - Sun, Jan 1, 2023 12:00:00 AM
                  </p>
                  <svg
                    class="h-4 w-4 shrink-0 transition-transform duration-200"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="m6 9 6 6 6-6"
                    />
                  </svg>
                </button>
              </h3>
              <div
                aria-labelledby="radix-:ra:"
                class="text-sm transition-all p-4"
                data-orientation="vertical"
                data-state="closed"
                hidden=""
                id="radix-:rb:"
                role="region"
                style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
              />
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
  ,
</DocumentFragment>
`;
